### ISSUE TITLE
Better dictionary printing

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#20638]
Consider this:
```
print([{"a": 1, "b": 2, "c": 3}])
print([{"a": 1, "b": 2}, {"c": 3}])
print([{"a": 1}, {"b": 2}, {"c": 3}])
```

Right now all of the output lines will look the same, so it's impossible to distinguish the actual structure of the arrays/dictionaries:
```
[(a:1), (b:2), (c:3)]
[(a:1), (b:2), (c:3)]
[(a:1), (b:2), (c:3)]
```

Dictionaries should show more than just pairs. Dunno why they don't print curly braces. It would look like this:
```
[{(a:1), (b:2), (c:3)}]
[{(a:1), (b:2)}, {(c:3)}]
[{(a:1)}, {(b:2)}, {(c:3)}]
```

For me this is a problem especially when working with json files.

**Comment by LikeLakers2** [comment]
I've always been a fan of how Ruby prints dictionaries:

```ruby
puts [{'a' => 1, 'b' => 2, 'c' => 3}]
# {"a"=>1, "b"=>2, "c"=>3}

puts [{'a' => 1, 'b' => 2}, {'c' => 3}]
# {"a"=>1, "b"=>2}
# {"c"=>3}

puts [{'a' => 1}, {'b' => 2}, {'c' => 3}]
# {"a"=>1}
# {"b"=>2}
# {"c"=>3}
```

Y'know, exactly as you type them in code? So I suggest something like this:
```gdscript
print([{"a": 1, "b": 2, "c": 3}])
# [{"a":1, "b":2, "c":3}]

print([{"a": 1, "b": 2}, {"c": 3}])
# [{"a":1, "b":2}, {"c":3}]

print([{"a": 1}, {"b": 2}, {"c": 3}])
# [{"a":1}, {"b":2}, {"c":3}]
```

**Comment by gotnospirit** [comment]
Hi,
https://github.com/godotengine/godot/blob/a71a5fc0c3f5d487d3f18c3b9e9a768052795e86/core/variant.cpp#L1479-L1483
We could change this code to mimic JSON::print or, better yet, replace the whole dictionary block with a call to JSON::print.
Furthermore, would it be more relevant to call JSON::print for every type it actually handle (NIL, BOOL, DICTIONARY, ...) and let custom code for the others (AABB, QUAT, ...)?

**Comment by KoBeWi** [comment]
Oooh, I was thinking about trying to contribute it myself, but couldn't find where it is defined.

No need for some weird JSON stuff, I like like how it is currently, only a pair of curly braces denoting dict beginning/end is missing. Right now it just prints a raw string of pairs, which leads to ambiguities I mentioned in OP.

**Comment by gotnospirit** [comment]
No problem!
Now you know where it is located.
I let you handle this :)

**Comment by akien-mga** [comment]
Fixed by #20856.
